packages/core/useStorage/index.ts

```diff
@@ -75,6 +75,13 @@ export interface UseStorageOptions<T> extends ConfigurableEventFilter, Configura
    */
   writeDefaults?: boolean
 
+  /**
+   * Merge the default value to the storage
+   *
+   * @default false
+   */
+  mergeDefaults?: boolean
```

Comment:
```suggestion
  mergeDefaults?: boolean | ((storage: T, defaults: T) => T)
```

We should allow them to provide a custom function for deep merge. We should also note that we only do shallow merge when this is set to `true`

Reply:
Nice suggest, I always learn something from your review .